geo/src/test/java/org/apache/carbondata/geo/QuadTreeClsTest.java

```diff
@@ -0,0 +1,317 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.carbondata.geo;
+
+import org.junit.*;
+import org.junit.rules.ExpectedException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.matchers.JUnitMatchers.containsString;
+
+
+/**
+ * QuadTreeClsTest Tester.
+ */
+public class QuadTreeClsTest {
+
+    @Rule
+    public ExpectedException exception = ExpectedException.none();
+
+    QuadTreeCls qtreee;
+
+    @Before
+    public void before() throws Exception {
+        qtreee = new QuadTreeCls(0, 0, 16, 16, 4 );
+    }
+
+    @After
+    public void after() throws Exception {
+        qtreee.clean();
+    }
+
+    @Test
+    public void testInsertPolygonFirstAndLastPoint() throws Exception {
```

Comment:
@VenuReddy2103 , @MarvinLitt : where is the end to end SQL based test case ?

we need test case to load CSV containing geo spatial data and query inpolygon udf and validate results.

Reply:
Yeah. I'll do in a new PR.